test/logflare/sql_test.exs

```diff
@@ -346,6 +346,77 @@ defmodule Logflare.SqlTest do
     end
   end
 
+  describe "clickhouse dialect" do
+    test "parser can handle tuple definitions" do
+      user = insert(:user)
+
+      for input <- [
+            "select tuple(1,2,3) as foo",
+            "select tuple(\'abc\')",
+            "select tuple(1, t.str_col) from values(\'str_col String\', (\'hello\'), (\'world\'), (\'test\')) t",
+            "SELECT tuple(t.str_col) AS abc from values(\'str_col String\', (\'hello\'), (\'world\'), (\'test\')) t"
+          ] do
+        assert {:ok, _v2} = Sql.transform(:ch_sql, input, user)
```

Comment:
something that i've noticed is that sometimes the parser can silently ignore some syntax without panicking, might want to make these tests a bit more resilient by making it spit the sql back out again from the ast

Reply:
if these panic on parsing then its fine